src/services/workshops/hooks.js

```diff
@@ -19,17 +19,40 @@ async function assignToOwner({ params: { user: owner }, app, result, service })
 async function checkXMLChanged(context) {
     const workshop = await context.app.service('workshops').get(context.id);
 
-    if (context.data.hasOwnProperty('xml')) {
+    if (context.data.hasOwnProperty('xml') && !context.data.hasOwnProperty('properties')) {
```

Comment:
To be honest, to avoid code repetition I would extract all the `if`s into separate function (ie. `contextChanged`) which would return either `true` or `false` and assign the returned value to `context.data.xmlChanged`. It will reduce side effects and shrink all the `context.data.xmlChanged` assignment blocks into one.